Docs/developer_guide/mrml_overview.md

```diff
@@ -100,14 +100,346 @@ The following methods on the MRML scene are used to manage Undo/Redo stacks:
+  * This string can be used in conjunction with the python method [`slicer.util.loadNodeFromFile`](https://slicer.readthedocs.io/en/latest/developer_guide/slicer.html#slicer.util.loadNodeFromFile)
+* Override [`QStringList extensions() const`](http://apidocs.slicer.org/master/classqSlicerFileReader.html#afb3187915977b2253d86634cc465b23d) to provide the extensions that can be read.
+  * Should be the same as the storage node since the reader uses the storage node.
+* Override [`bool load(const IOProperties& properties)`](http://apidocs.slicer.org/master/classqSlicerFileReader.html#ac9acb878cd8adcc426e9a7a9edac15df). This is the function that actually loads the node from the file into the scene.
+
+:::{important}
+
+The reader is not a VTK object, like the previous objects discussed. It is actually a QObject, so we follow Qt guidelines. One such guideline is the [D-Pointer pattern](https://wiki.qt.io/D-Pointer), which is recommended for use.
+
+:::
+
+#### The writer
+
+The writer is the companion to the reader, so, similar to the reader, it does not implement the actual writing of files, but rather it uses the storage node. Its existence is necessary to use 3D Slicer’s built in saving facilities, such as the save button.
+
+Files:
+
+* \<Extension>/\<Module>/qSlicer\<MyCustomType>Writer.h
+* \<Extension>/\<Module>/qSlicer\<MyCustomType>Writer.cxx
+
+Key points:
+
+* Naming convention for class: `qSlicer<MyCustomType>Writer`
+* Inherits from [`qSlicerNodeWriter`](http://apidocs.slicer.org/master/classqSlicerNodeWriter.html).
+* See the [reader](#the-reader) for information on defining and constructing Qt style classes.
+* Override [`QStringList extensions(vtkObject* object) const`](http://apidocs.slicer.org/master/classqSlicerNodeWriter.html#aa2d7322c22d3d5fa7b9ef2843948b31a) to provide file extensions that can be written to.
+  * File extensions may be different, but don’t have to be, for different data nodes that in the same hierarchy (e.g. Markups Curve and Plane could reasonably require different file extensions, but they don’t).
+* Override [`bool write(const qSlicerIO::IOProperties& properties)`](http://apidocs.slicer.org/master/classqSlicerNodeWriter.html#aa7e3af9bf485b46735131e6363223ad8) to do the actual writing (by way of a storage node, of course).
+
+#### The subject hierarchy plugin
+
+A convenient module in 3D Slicer is the Data module. It brings all the different data types together under one roof and offers operations such as cloning, deleting, and renaming nodes that work regardless of the node type. The Data module uses the Subject Hierarchy, which is what we need to plug into so our new node type can be seen in and modified by the Data module.
+
+Files:
+
+* \<Extension>/\<Module>/SubjectHierarchyPlugins/qSlicerSubjectHierarchy\<MyCustomType>Plugin.h
+* \<Extension>/\<Module>/SubjectHierarchyPlugins/qSlicerSubjectHierarchy\<MyCustomType>Plugin.cxx
```

Comment:
```
|-- <Extension>
       |-- <Module>
              |-- SubjectHierarchyPlugins
                    |-- qSlicerSubjectHierarchy<MyCustomType>Plugin.h
                    |-- qSlicerSubjectHierarchy<MyCustomType>Plugin.cxx
```

Reply:
Updated to suggested file structure formatting.